### PR TITLE
Knot theory fixes

### PR DESCRIPTION
**Pull request by mmatera**

### CODE REVIEW

**Comment by rocky** [comment]
@mmatera  think I understand how to write the type table-lookup code I was describing in a rough way. 

What would help though is are some useful tests that come up.

File: mathics/builtin/files.py [hunk 1]
```diff
@@ -712,7 +715,17 @@ def apply(self, channel, types, evaluation, options):
 
         # Wrap types in a list (if it isn't already one)
         if not types.has_form("List", None):
+            if types.get_head_name() == "System`Hold":
```
Comment (rocky) [inline mathics/builtin/files.py:718]: This seems hoaky since the speck seems to allow any function and not just `Hold[]` right?

**Review by rocky — CHANGES_REQUESTED** [review]
Please let's start splitting this up into the individual pieces and create separate PRs for this.

For example Setting System`Notebooks is a thing in of itself and not even needed for Feyncalc.

That could be merged in by itself.

The problem with these growing grab-bag PRs is that they are impossible to review and the quality can suffer because we can't focus on individual features. 

And then they are continually merged (when rebasing instead of merging would lead to more comprehensible results)

File: mathics/builtin/strings.py [hunk 2]
```diff
@@ -829,6 +829,9 @@ class StringSplit(Builtin):
         <dd>splits $s$ at the delimiter $d$.
     <dt>'StringSplit[$s$, {"$d1$", "$d2$", ...}]'
         <dd>splits $s$ using multiple delimiters.
+    <dt>'StringSplit[{$s_1$, $s_2, $\ldots$}, {"$d1$", "$d2$", ...}]'
```
Comment (rocky) [inline mathics/builtin/strings.py:855]: Note that ldots doesn't render in the Django interface.

**Comment by mmatera** [comment]
Spli

> Please let's start splitting this up into the individual pieces and create separate PRs for this.
> 
> For example Setting System`Notebooks is a thing in of itself and not even needed for Feyncalc.
> 
> That could be merged in by itself.
> 
> The problem with these growing grab-bag PRs is that they are impossible to review and the quality can suffer because we can't focus on individual features.
> 
> And then they are continually merged (when rebasing instead of merging would lead to more comprehensible results)

Splitted on #1219, #1209 #1216, #1217

**Comment by rocky** [comment]
Superceded by other PRs.

**Pull request closed without merging**